assets/js/Containers/FrevaGPT/components/AnswerComponent.js

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Col, Card, Spinner } from "react-bootstrap";
+
+import Markdown from "react-markdown";
+
+function AnswerComponent(props) {
+  function renderAnswer(props) {
+    switch (props.variant) {
+      case "Assistant":
+        return (
+          <Col md={{ span: 10, offset: 0 }}>
+            <Card className="shadow-sm card-body border-0 border-bottom mb-3 bg-light">
+              <Markdown>{props.content}</Markdown>
+            </Card>
+          </Col>
+        );
+      case "Code":
+      case "CodeBlock":
+        return (
+          <Col md={{ offset: 0 }}>
+            <Card className="shadow-sm card-body border-0 border-bottom mb-3 bg-light">
+              <span>
+                <Spinner size="sm" />
+                Analyzing code...
+              </span>
+            </Card>
+          </Col>
+        );
+      default:
```

Comment:
If you click on the example `Make a zonal mean plot of precipitation for the average between 1961 and 1990 as well as 1991 and 2020`, there is a pause after the first Code Output where no Spinner is visible and the system does also not create any visible output despite being not finished. It seems that during this phase the API is producing `ServerHint`s. I am not sure how you want to deal with it, but for me it seems that we want to see another Spinner when the `ServerHint`s are popping up.